src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs

```diff
@@ -0,0 +1,111 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.ServiceFabric.Services.Remoting.V2
+{
+    using System.Diagnostics;
+#if DotNetCoreClr
+    /// <summary>
+    /// Class ActivityIdLogicalCallContext.
+    /// </summary>
+    internal class ActivityIdLogicalCallContext
+    {
+        /// <summary>
+        /// Creates the activity.
+        /// Note : You still need to start it.
+        /// </summary>
+        /// <param name="activityMessage">The activity message.</param>
+        /// <returns>Activity.</returns>
+        internal Activity CreateW3CActivity(string activityMessage = "Create new Activity")
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+            var activity = new Activity(activityMessage);
+            return activity;
+        }
+
+        internal Activity StartActivity(IServiceRemotingRequestMessage requestMessage, string activityMessage = "Start new Activity")
+        {
+            var activity = new Activity(activityMessage);
+            string parentId = null;
+            if (requestMessage.GetHeader().ActivityIdParent != null)
+            {
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+                Activity.ForceDefaultIdFormat = true;
+                parentId = requestMessage.GetHeader().ActivityIdParent;
+            }
+            else
+            {
+                parentId = requestMessage.GetHeader().ActivityRequestId;

```

Comment:
In both the cases, we are setting the parent. Is HttpListener code is also doing the same? If ActicityIdParent is used to identify W3C format, we can probably send the format itself?

Reply:
Http does the same

Reply:
Link to reference -> https://github.com/vancem/AspNetCore/blob/38ac3da95426132c7f90bfc88e02e308d151eea3/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#:~:text=activity.SetParentId(requestId)%3B